trade-module/src/main/java/io/edpn/backend/trade/adapter/config/KafkaConfig.java

```diff
@@ -53,8 +62,24 @@ public ConsumerFactory<String, JsonNode> consumerFactory(String groupId) {
         }
 
         @Bean(name = "tradeModuleKafkaListenerContainerFactory")
-        public ConcurrentKafkaListenerContainerFactory<String, JsonNode> kafkaListenerContainerFactory(EddnJsonKafkaConsumerConfig kafkaConfig) {
-            return kafkaConfig.kafkaListenerContainerFactory("tradeModule");
+        public ConcurrentKafkaListenerContainerFactory<String, JsonNode> kafkaListenerContainerFactory(EddnJsonKafkaConsumerConfig kafkaConfig, @Qualifier("tradeKafkaErrorHandler")CommonErrorHandler errorHandler) {
+            return kafkaConfig.kafkaListenerContainerFactory("tradeModule", errorHandler);
+        }
+
+        @Bean(name = "tradeKafkaErrorHandler")
+        public DefaultErrorHandler errorHandler(
+                @Value(value = "${trade.kafka.backoff.interval:1000}") final int interval,
+                @Value(value = "${trade.kafka.backoff.multiplier:2000}") final int multiplier,
+                @Value(value = "${trade.kafka.backoff.max_interval:32000}") final int maxInterval
+        ) {
+            ExponentialBackOff backOff = new ExponentialBackOff();
+            backOff.setInitialInterval(interval);
+            backOff.setMultiplier(multiplier);
+            backOff.setMaxInterval(maxInterval);
+            return new DefaultErrorHandler((consumerRecord, exception) -> {
+                // TODO should there be extra logic to execute when all the retry attempts are exhausted, or do we just drop the message?
+                log.error("A kafka message from topic '${}' could not be processed after multiple retries: '${}'", consumerRecord.topic(), consumerRecord.value(), exception);
```

Comment:
Are the dollar signs a mistake, or a formatting notation I'm just unaware of? AFAIK, SLF4J formatting anchor is simply '{}', no dollar sign.